python/lsst/sims/utils/coordinateTransformations.py

```diff
@@ -103,7 +62,194 @@ def calcLmstLast(mjd, longRad):
     last %= 24.
     return lmst, last
 
-def raDecToAltAzPa(raRad, decRad, longRad, latRad, mjd):
+
+def horizontalFromEquatorial(ra, dec, mjd, longitude, latitude):
+    """
+    Converts from equatorial to horizon coordinates
+
+    @param [in] ra is in radians
+
+    @param [in] dec is declination in radians
+
+    @param [in] mjd is the date
+
+    @param [in] longitude is the site longitude in radians
+    (positive to the east of the prime meridian)
+
+    @param [in] latitude is the site latitude in radians
+
+    @param [out] returns elevation angle and azimuth in that order (radians)
+
+    """
+
+    hourAngle = calcLmstLast(mjd, longitude)[1]*(2.0*numpy.pi/24.0) - ra
+
+    _de2hOutput=palpy.de2h(hourAngle, dec,  latitude)
+
+    #return (altitude, azimuth)
+    return _de2hOutput[1], _de2hOutput[0]
+
+
+def galacticFromEquatorial(ra, dec):
+    '''Convert RA,Dec (J2000) to Galactic Coordinates
+
+    All angles are in radians
+
+    @param [in] ra is right ascension in radians, either a float or a numpy array
+
+    @param [in] dec is declination in radians, either a float or a numpy array
+
+    @param [out] gLong is galactic longitude in radians
+
+    @param [out] gLat is galactic latitude in radians
+    '''
+
+    if isinstance(ra, numpy.ndarray):
+        gLong, gLat = palpy.eqgalVector(ra, dec)
+    else:
+        gLong, gLat = palpy.eqgal(ra, dec)
+
+    return gLong, gLat
+
+
+def equatorialFromGalactic(gLong, gLat):
+    '''Convert Galactic Coordinates to RA, dec (J2000)
+
+    @param [in] gLong is galactic longitude in radians, either a float or a numpy array
+
+    @param [in] gLat is galactic latitude in radians, either a float or a numpy array
+
+    @param [out] ra is right ascension in radians
+
+    @param [out] dec is declination in radians
+    '''
+
+    if isinstance(gLong, numpy.ndarray):
+        ra, dec = palpy.galeqVector(gLong, gLat)
+    else:
+        ra, dec = palpy.galeq(gLong, gLat)
+
+    return ra, dec
+
+
+def cartesianFromSpherical(longitude, latitude):
+    """
+    Transforms between spherical and Cartesian coordinates.
+
+    @param [in] longitude is the input longitudinal coordinate
+
+    @param [in] latitude is the input latitudinal coordinate
+
+    @param [out] a list of the (three-dimensional) cartesian coordinates on a unit sphere
+
+    All angles are in radians
+    """
+
+    cosDec = numpy.cos(latitude)
+    return numpy.array([numpy.cos(longitude)*cosDec,
+                      numpy.sin(longitude)*cosDec,
+                      numpy.sin(latitude)])
+
+
+def sphericalFromCartesian(xyz):
+    """
+    Transforms between Cartesian and spherical coordinates
+
+    @param [in] xyz is a list of the three-dimensional Cartesian coordinates
+
+    @param [out] returns longitude and latitude
+
+    All angles are in radians
+    """
+
+    rad = numpy.sqrt(xyz[:][0]*xyz[:][0] + xyz[:][1]*xyz[:][1] + xyz[:][2]*xyz[:][2])
```

Comment:
Why don't you just do numpy.sqrt(xyz[0]*xyz[0] + xyz[1]*xyz[1] + xyz[2]*xyz[2])?